packages/discord.js/typings/index.d.ts

```diff
@@ -3793,6 +3793,30 @@ export interface ApplicationCommandChannelOption extends BaseApplicationCommandO
   channelTypes?: ChannelType[];
 }
 
+export interface ApplicationCommandRoleOptionData extends BaseApplicationCommandOptionsData {
+  type: ApplicationCommandOptionType.Role;
+}
+
+export interface ApplicationCommandRoleOption extends BaseApplicationCommandOptionsData {
+  type: ApplicationCommandOptionType.Role;
+}
+
+export interface ApplicationCommandUserOptionData extends BaseApplicationCommandOptionsData {
+  type: ApplicationCommandOptionType.User;
+}
+
+export interface ApplicationCommandUserOption extends BaseApplicationCommandOptionsData {
+  type: ApplicationCommandOptionType.User;
+}
+
+export interface ApplicationCommandMentionableOptionData extends BaseApplicationCommandOptionsData {
+  type: ApplicationCommandOptionType.Mentionable;
+}
+
+export interface ApplicationCommandMentionableOption extends BaseApplicationCommandData {
```

Comment:
```suggestion
export interface ApplicationCommandMentionableOption extends BaseApplicationCommandOptionsData {
```

Reply:
I'm curious to know as to why a command option extends from a data type meant for commands.

I admit I'm not familar enough with d.js source code to see how this works out. Under normal circumstances, I'd have either started a discussion or rejected this request. But considering you're a maintainer of this repo, I've accepted this suggestion.